### PR TITLE
Ndbex 95822 logging and email vre

### PR DESCRIPTION
**Pull request by evansmith**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): /NO*
- Added a monitor, inheriting from the ZSF monitor, to log when retries are exhausted in VRE. Additionally added a VA Notify call.
- VA Benefits and Claims
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

[- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*](https://github.com/department-of-veterans-affairs/va.gov-team/issues/93112)
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [x] *New code is covered by unit tests*
- Old behavior we just logged and increased statsd. This uses a more formal library that adds a ZSF notification and includes an email to the Veteran.
- Submit a 1900 claim that fails (can be done in staging). Confirm in the logs that an email went out and that logs were created.
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by micahaspyr** [comment]
Besides the above comment this looks good once the linting is fixed

File: lib/vre/monitor.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    def initialize
+      super('vre-application')
+    end
+
+    def track_submission_exhaustion(msg)
+      additional_context = {
+        message: msg
+      }
+      log_silent_failure(additional_context, nil, call_location: caller_locations.first)
```
[the remark this answers is not in the document]
Comment (mjknight50) [reply lib/vre/monitor.rb:24]: The silent part refers to the veteran, not to us.  Confusing...
Silent to veteran

File: app/sidekiq/vre/submit1900_job.rb [hunk 2]
```diff
@@ -11,10 +13,24 @@ class Submit1900Job
...
 
     sidekiq_retries_exhausted do |msg, _ex|
-      Rails.logger.error(
-        "Failed all retries on VRE::Submit1900Job, last error: #{msg['error_message']}"
-      )
-      StatsD.increment("#{STATSD_KEY_PREFIX}.exhausted")
+      monitor = VRE::Monitor.new
+      monitor.track_submission_exhaustion(msg)
+
+      if Flipper.enabled?(:vre_trigger_action_needed_email)
```
Comment (micahaspyr) [inline app/sidekiq/vre/submit1900_job.rb:19]: It would be best to create a method for everything in this ' if Flipper.enabled?' block for readability.

**Review by micahaspyr — CHANGES_REQUESTED** [review]
Just the one change

File: lib/vre/monitor.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (micahaspyr) [reply lib/vre/monitor.rb:24]: We should add some comments for clarity then.

[the remark this answers is not in the document]
Comment (mjknight50) [reply lib/vre/monitor.rb:26]: It is an ongoing source of confusion.  Agreed that comments would help.

**Review by micahaspyr — APPROVED** [review]
LGTM

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**